eras/shelley/impl/src/Cardano/Ledger/Shelley/Genesis.hs

```diff
@@ -129,6 +149,71 @@ emptyGenesisStaking =
       sgsStake = mempty
     }
 
+-- | Unlike @'NominalDiffTime'@ that supports @'Pico'@ precision, this type
+-- only supports @'Micro'@ precision.
+newtype NominalDiffTimeMicro = NominalDiffTimeMicro Micro
+  deriving (Show, Eq, Generic)
+  deriving anyclass (NoThunks)
+  deriving newtype (Ord, Num, Fractional, Real, ToJSON, FromJSON)
+
+deriving instance Generic Micro
+
+deriving anyclass instance NoThunks Micro
+
+instance ToCBOR NominalDiffTimeMicro where
+  toCBOR = encodeNominalDiffTimeMicro
+
+encodeNominalDiffTimeMicro :: NominalDiffTimeMicro -> Encoding
+encodeNominalDiffTimeMicro (NominalDiffTimeMicro micro) =
+  encodeInteger . toMicroseconds $ micro
+  where
+    toMicroseconds t =
+      numerator (toRational t * toRational (resolution $ Proxy @E6))
+
+instance FromCBOR NominalDiffTimeMicro where
+  fromCBOR = decodeNominalDiffTimeMicro
+
+decodeNominalDiffTimeMicro :: Decoder s NominalDiffTimeMicro
+decodeNominalDiffTimeMicro =
+  NominalDiffTimeMicro . fromRational . (% 1_000_000) <$> decodeInteger
```

Comment:
I think this one is wrong, it should be:
```suggestion
  NominalDiffTimeMicro . MKFixed <$> decodeInteger
```

Reply:
Same here as for the `encode`. This could be changed to `deriving newtype (FromCBOR)` as we added the instance for `Micro` to `Cardano.Ledger.Binary.Decoding.FromCBOR`. :+1: